forge-game/src/main/java/forge/game/ability/effects/TextBoxExchangeEffect.java

```diff
@@ -0,0 +1,163 @@
+package forge.game.ability.effects;
+
+import com.google.common.collect.Lists;
+import com.google.common.collect.Table.Cell;
+import com.google.common.collect.HashBasedTable;
+import com.google.common.collect.Table;
+
+import forge.game.Game;
+import forge.game.ability.SpellAbilityEffect;
+import forge.game.card.Card;
+import forge.game.card.CardState;
+import forge.game.event.GameEventCardStatsChanged;
+import forge.game.keyword.KeywordInterface;
+import forge.game.card.CardTraitChanges;
+import forge.game.replacement.ReplacementEffect;
+import forge.game.spellability.SpellAbility;
+import forge.game.staticability.StaticAbility;
+import forge.game.trigger.Trigger;
+
+import java.util.List;
+
+/**
+ * Exchanges text boxes between two creatures.
+ */
+public class TextBoxExchangeEffect extends SpellAbilityEffect {
+    @Override
+    protected String getStackDescription(final SpellAbility sa) {
+        final List<Card> tgtCards = getTargetCards(sa);
+        Card c1;
+        Card c2;
+        if (tgtCards.size() == 1) {
+            c1 = sa.getHostCard();
+            c2 = tgtCards.get(0);
+        } else {
+            c1 = tgtCards.get(0);
+            c2 = tgtCards.get(1);
+        }
+        return c1 + " exchanges text box with " + c2 + ".";
+    }
+
+    @Override
+    public void resolve(final SpellAbility sa) {
+        final List<Card> tgtCards = getTargetCards(sa);
+        final Card host = sa.getHostCard();
+        if (tgtCards.size() < 2) {
+            return;
+        }
+
+        final Card c1 = tgtCards.get(0);
+        final Card c2 = tgtCards.get(1);
+
+        // snapshot the original text boxes before modifying
+        final TextBoxData data1 = captureTextBoxData(c1);
+        final TextBoxData data2 = captureTextBoxData(c2);
+
+        final Game game = host.getGame();
+        final long ts = game.getNextTimestamp();
+
+        swapTextBox(c1, data2, ts);
+        swapTextBox(c2, data1, ts);
+
+        game.fireEvent(new GameEventCardStatsChanged(c1));
+        game.fireEvent(new GameEventCardStatsChanged(c2));
+    }
+
+    private static void swapTextBox(final Card to, final TextBoxData from, final long ts) {
+        to.getChangedCardTraitsByText().clear();
+        to.clearChangedCardKeywords(false);
+        to.copyChangedTextFrom(from.textHolder);
```

Comment:
Ok, so the reason why you might have thought you needed to copy the previous text changes extra is that normally `CardTraitBase.copyHelper` should not apply them again

However we really shouldn't mess with it manually considering there's also a duration-based exchange already printed (_Exchange of Words_)

`SpellAbility.copy()` is already overloaded to support that, probably that should be done for the other traits too 🤔